include/fmt/ranges.h

```diff
@@ -291,10 +291,10 @@ struct formatter<TupleT, Char, enable_if_t<fmt::is_tuple_like<TupleT>::value>> {
         }
         out = detail::copy(formatting.delimiter, out);
       }
-      out = format_to(out,
-                      detail::format_str_quoted(
-                          (formatting.add_delimiter_spaces && i > 0), v),
-                      v);
+      out = vformat_to(out,
+                       detail::format_str_quoted(
+                           (formatting.add_delimiter_spaces && i > 0), v),
+                       make_format_args<FormatContext>(v));
```

Comment:
Here and below: it would be better to refactor this to use compile string and `format_to`.

Reply:
I considered that but given that the format string used appeared to be determined at run-time I wasn't sure it was possible. I'll give it a go.

Reply:
Looking at this more I'm reaching the same conclusion - `formatting` can't be used from within `FMT_STRING` - doing so causes 
```
/home/walter.gray/development/walrus/fmt/include/fmt/ranges.h:371:29: error: use of non-static data member 'formatting' of 'formatter<T, Char, enable_if_t<fmt::is_range<T, Char>::value && (has_formatter<detail::value_type<T>, format_context>::value || detail::has_fallback_formatter<detail::value_type<T>, format_context>::value)> >' from nested type 'FMT_COMPILE_STRING'
                           (formatting.add_delimiter_spaces && i > 0), *it)),```
                           
I couldn't have a function that returned one of several format strings, and I don't think I could have a function that returned a constexpr string that would work with FMT_STRING either - 
```
constexpr const char* getString() {
    return "{} bananna";
}
int main(int argc, char** argv) {
    fmt::print(FMT_STRING(getString()), "fried");
}
```
for example will not compile in godbolt. I also cant return the result of FMT_STRING since they'd be different types between branches. My best guess for how to do this would be to have a wrapping function that used template specialization on a bool for add_delimiter_spaces and runtime switching on i > 0 and called format_to with the appropriate FMT_STRING directly. Am I missing an option here or should I try that?

Reply:
Let's use `write` instead of `format_to` here too.